### ISSUE TITLE
GTX 1070 issues with High Sierra 10.13.1 and 10.13.2, 60Hz and 4K via HDMI 2.0

### ISSUE THREAD

**Issue by mutatioveritas** [Floris497/mac-pixel-clock-patch-V2#246]
I'm having the black screen issues after a fresh install of HS and installation of nVidia drivers on two installs on the same system (separate drives), trying to use HDMI 2.0, which had been successful with your earlier patches in El Capitan. I'm using iMac 14,2 which is supposed to not have these issues or require Lilu or nVidiaGraphicsFixup. I have gotten a poor workaround where I have Lilu, nVidiaGraphicsFixup, and CoreDisplayFixup in my Other Clover folder, which loads the drivers but gives me the black screen. I tried what others have done and used a remote desktop app to alter the display settings. My Samsung 40" JU7100 seems to be seen as a TV. Part of getting the black screen to resolve is turning off "overscan" in the display settings, which tells me that the system is seeing it as a TV rather than a pixel-for-pixel monitor. With overscan unchecked, I get a display that doesn't span my screen regardless of the resolution I try. I get a black border around the screen.

As a side note, I used the pixel clock patcher and later the IOKit patcher on 10.11.x to give me a flawless setup with the same TV, albeit with a GTX 960. I upgraded the GPU along with the intention to upgrade the system to HS. 

I upgraded the 10.13.1 install to 10.13.2 as the CoreDisplay patch wasn't taking, got it to fully run in 10.13.2, but I'm not seeing any change over the use of the kexts. Same result. Display seems to be treated as a TV rather than a monitor.

Any ideas?

**Comment by Floris497** [comment]
HDMI2.0 is a no go right now.. sorry 
Does 30Hz work though?
Displayport is the answer..
Hm.. do you still have the 960? I wonder if that one would work.. but I doubt it

**Comment by mutatioveritas** [comment]
Looks like I will have to try something like the 3D club adapter?

**Comment by Floris497** [comment]
I'll read thoroughly tomorrow..

**Comment by Floris497** [comment]
Hm.. I'm actually not sure at this point.. if HDMI2.0 works at all even with an adapter..

**Comment by Floris497** [comment]
Okay.. but.. it IS a TV right?

**Comment by Floris497** [comment]
Okay let me check a few things tomorrow! And I'll come with a more useful answer

**Comment by mutatioveritas** [comment]
Yes, it is formally a TV.

**Comment by mutatioveritas** [comment]
Correct, it is a TV. Had similar issues with the GTX 960 and 10.11.x without the patches, weird resolutions being offered, like 2560x1350 (rather than 1440). With the patches in 10.11.x, the TV appeared to be treated as a monitor, with all HiDPI resolutions available and properly rendered. Will go down and plug in the 960 and see if that makes a difference or not.

**Comment by mutatioveritas** [comment]
Bumped into a slightly better workaround. The TV has a UHD color setting. If enabled, it allows 4:4:4 chroma at 4K @ 60Hz (as a monitor should), which is what I had it set for with 10.11.x, the GTX 960, and the IOKit patcher. As noted, it worked great and without issue. I was checking the TV settings to see if perhaps the UHD setting might have gotten switched off to the default setting due to automatic firmware updates. It was still on, but I opted to switch it off to see if it impacted the display. The resulting image appeared to be the same, with the black border around the screen. On a lark, I decided to click on the overscan button, which then rendered the HiDPI resolution properly on the screen. The only problems being, I lost 4:4:4 chroma and 60Hz. With the black border settings, I got 24Hz. With proper screen space being used with switching the TV setting, I'm up to 30Hz, a slight improvement, but I still can't enable 60Hz and, as noted, I've lost 4:4:4 chroma. 

Again, my best hunch is that there is something in the CoreDisplay settings that is getting stuck on the TV settings. If I recall correctly, I never saw the overscan option in the display settings with 10.11.x, GTX 960, and IOKit Patcher. Now that I've gotten a stop gap workaround, I will likely downgrade one of my drives to 10.11.x and restore from a Time Machine backup, plug in the GTX 960, and double check the display settings with the IOKit patcher in the mix. Crossing my fingers that maybe we can get 60Hz in the mix!

BTW, the adapter I was referring to was a Displayport to HDMI 2.0 adapter.

**Comment by mutatioveritas** [comment]
Any idea if there will be a 4K 60Hz HDMI 2.0 solution like there was with the iOKit patch?

**Comment by limi** [comment]
I'm on a MacPro5,1 running 10.13.2 (17C88) with an Nvidia 1050Ti card, and was able to get 4K@60hz without installing the patch, using an Accel DisplayPort → HDMI 2.0 adapter. I was hoping to unlock a second 4K@60hz output via the HDMI or DVI port using this patch, but can only get 30hz out of the HDMI port. 

Should this be theoretically possible, or am I trying to do something that is not known to work? I'm also happy to downgrade to Sierra instead of High Sierra if that would help matters.

Thank you for your work on this, apologies if my questions are covered elsewhere — the whole situation is a bit confusing.

**Comment by limi** [comment]
(Looks like I need to downgrade to El Capitan, not Sierra — I might need the IOKit patch to make this work as opposed to the CoreDisplay patch?)

**Comment by LonnieCoffman** [comment]
I have 4k@60hz working with high sierra with a 48" Samsung TV (JU6700) through HDMI on a GTX970, although if I shut off the TV and turn in back on I end up with a black screen and have to reboot.  I was able to get 4k@60hz working by turning off CSM support in my bios (Gigabyte).  To display this option I had to set "Windows 8/10 Features:" to Windows 8/10.

As a note, I am on the 17,1 system definition and am not using Lilo or nVidiaGraphicsFixup, but rather the script from the second post of this thread: [https://www.tonymacx86.com/threads/...s-high-sierra-10-13-1-378-10-10-10-20.236101/](url)

Until I can find a solution to the black screen when power cycling the monitor I am using a Club3D DP to HDMI adapter which works great.

**Comment by mutatioveritas** [comment]
Lonnie, that is interesting to hear. I don't know that I saw that script in all my searches. Any luck with 10.13.2 and that script? It seems Mac OS and nVidia support is a hodgepodge. I've seen some say that they could get things working with CSM enabled, so I might have to go back in and tweak some settings. By default, all my earlier unsuccessful efforts were with CSM disabled. This was all with the JU6700 set for UHD color I imagine?

Lastly, I am using the workaround as well with the Club3D adapter for 60Hz the TV's pipes fully open with UHD color enabled. The only issue I'm seeing is that when I check on the display specifics in the system report, it notes 24-bit color and not 32. Are you seeing that as well? When I boot into Windows I see full 32-bit color indicated at 4K 60Hz.

**Comment by mutatioveritas** [comment]
I did tweak the CSM setting and I think I'm seeing what you had indicated, Lonnie. I have Windows 8/10  indicated in the BIOS, with CSM disabled. I can now boot into OS X @ 4K 60Hz over HDMI 2.0! I did, however, put the system to sleep and turned the TV off. When I woke the computer from sleep, the display did not re-engage for whatever reason, so I am getting the black screen as well with HDMI. I returned my Club3D adapter. While it worked for booting into 4K @ 60Hz, wake from sleep gave me a distorted screen that is useless. If I have to fully shut down the system with either the HDMI or DP/HDMI connection to avoid the issues after the TV is turned off and back on, I'll just stick with HDMI.

**Comment by oscarbg** [comment]
Hi @mutatioveritas,
I'm on 10.13.3 with 387 driver not 378 drivers from 10.13.1 or 10.13.2..
so I have GTX970 and a Gigabyte motherboard.. are you telling me all I need to do is "have Windows 8/10 indicated in the BIOS, with CSM disabled" or have to apply the patch @LonnieCoffman said also?
is this patch compatible with 10.13.3? 
thanks..

**Comment by mutatioveritas** [comment]
Hi @oscarbg,  The cocktail that I would suggest trying before the patch would be the CoreDisplayFixup (CDF), nVidiaGraphicsFixup, and Lilu kexts in your "Other" folder in your Clover EFI partition. As I understand it, CDF is supposed to integrated @Floris497's relevant patches. Try those with the Windows 8/10 setting enabled and CSM disabled and see what happens. I've used Clover to try loading the system without any one of those three kexts and it won't fly if one of them is disabled from loading. If those work, then you won't have to worry about patching system files.

**Comment by mutatioveritas** [comment]
@LonnieCoffman, Maybe I got a dud with the Club 3D adapter. I got the most recent edition and while it did allow the system to boot and render HiDPI resolutions, it was a bust on waking up. The screen would be all distorted and unusable. I returned the adapter. You're saying that your adapter properly renders the display after waking from sleep?
